apps/voxelize/voxelize.cpp

```diff
@@ -129,30 +128,27 @@ class Voxelize : public CommandLineApplication
 
     void sample()
     {
-        const ::fivox::URIHandler params( getURI( ));
-        ImageSourcePtr source = params.newImageSource< float >();
-        ::fivox::EventSourcePtr loader = source->getFunctor()->getSource();
-        const fivox::AABBf& bbox = loader->getBoundingBox();
-
-        const fivox::Vector3f& extent( bbox.getSize() +
-                                       params.getExtendDistance() * 2.f );
+        ::fivox::URI uri = getURI();
 
-        size_t size;
+        // for compatibility
         if( _vm.count( "size" ))
-            size = _vm["size"].as< size_t >();
-        else
-        {
-            const fivox::Vector3f& sizeVoxels = extent * params.getResolution();
-            size = (size_t)std::ceil( sizeVoxels.find_max( ));
-        }
+            uri.addQuery( "size", std::to_string( _vm["size"].as< size_t >( )));
```

Comment:
Same train of though as I had, but can't be lexical_casted from the commandline if not specified as string... :) Hence, falling back was working.